examples/flutter_gallery/lib/demo/material/bottom_app_bar_demo.dart

```diff
@@ -0,0 +1,571 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+class BottomAppBarDemo extends StatefulWidget {
+  static const String routeName = '/material/bottom_app_bar';
+
+  @override
+  State createState() => new _BottomAppBarDemoState();
+}
+
+class _BottomAppBarDemoState extends State<BottomAppBarDemo> {
+
+  // The index of the currently-selected _FabLocationConfiguration.
+  int fabLocationIndex = 1;
+
+  static const List<_FabLocationConfiguration> _fabLocationConfigurations = const <_FabLocationConfiguration>[
+    const _FabLocationConfiguration('Center, undocked above the bottom app bar', _BabMode.CENTER_FAB, FloatingActionButtonLocation.centerFloat),
+    const _FabLocationConfiguration('End, undocked above the bottom app bar', _BabMode.END_FAB, FloatingActionButtonLocation.endFloat),
+    const _FabLocationConfiguration('Center, docked to the bottom app bar', _BabMode.CENTER_FAB, const _CenterDockedFloatingActionButtonLocation()),
+    const _FabLocationConfiguration('End, docked to the bottom app bar', _BabMode.END_FAB, const _EndDockedFloatingActionButtonLocation()),
+    const _FabLocationConfiguration('Start, docked to the top app bar', _BabMode.CENTER_FAB, const _TopStartFloatingActionButtonLocation()),
+  ];
+  
+  // The index of the currently-selected _FabShapeConfiguration.
+  int fabShapeIndex = 0;
+
+  final List<_FabShapeConfiguration> _fabShapeConfigurations =  <_FabShapeConfiguration>[
+      const _FabShapeConfiguration('None', null),
+      new _FabShapeConfiguration('Circular', 
+        new Builder(builder: (BuildContext context) {
```

Comment:
It's a shape that this Build is needed (to create a context from which the Scaffold can be found).

It would be slightly preferable to give the Scaffold a global _scaffoldKey and have _showSnackbar() dispense with the context argument and ` _scaffoldKey.currentState.showSnackBar()`.

This is a common pattern in the gallery demos.

Reply:
Done, thanks for the suggestion.